### PR TITLE
Signature verification

### PR DESCRIPTION
**Pull request by aligeti**
https://storjlabs.atlassian.net/browse/V3-683

### CODE REVIEW

File: pkg/bwagreement/server.go [hunk 1]
```diff
@@ -100,6 +107,10 @@ func (s *Server) BandwidthAgreements(stream pb.Bandwidth_BandwidthAgreementsServ
 		case <-ctx.Done():
 			return nil
 		case agreement := <-ch:
+			err = s.verifySignature(ctx, agreement)
```
Comment (aleitner) [inline pkg/bwagreement/server.go]: Combine lines 110 and 111

File: pkg/bwagreement/server.go [hunk 2]
```diff
@@ -100,6 +107,10 @@ func (s *Server) BandwidthAgreements(stream pb.Bandwidth_BandwidthAgreementsServ
...
+			if err != nil {
+				return err
```
Comment (aleitner) [inline pkg/bwagreement/server.go]: It might be better to just log this error and `continue`

File: pkg/bwagreement/server.go [hunk 3]
```diff
@@ -117,3 +128,21 @@ func (s *Server) GetBandwidthAllocations(ctx context.Context) (rows []*dbx.Bwagr
 	rows, err = s.DB.All_Bwagreement(ctx)
 	return rows, err
 }
+
+func (s *Server) verifySignature(ctx context.Context, ba *pb.RenterBandwidthAllocation) error {
```
Comment (aleitner) [inline pkg/bwagreement/server.go]: Do we also want to verify the payer bandwidth signature? @jtolds

Comment (aligeti) [reply pkg/bwagreement/server.go]: added the check

File: pkg/bwagreement/server.go (hunk 2 above)
Comment (aligeti) [reply pkg/bwagreement/server.go]: Done

File: pkg/bwagreement/server.go (hunk 1 above)
Comment (aligeti) [reply pkg/bwagreement/server.go]: Done

**Pull request merged**